### DISCUSSION TITLE
Add nested typed collections

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#10022] [Scripting]
Currently, in Godot v4.2.2 if you try to insert the code
```gdscript
var variable : Array[Array[bool]]
``` 
into a script, you will get back an error that says "Nested typed collections are not supported." I was wondering if anyone knew enough about the engine to implement this, and if it should be added.

**Comment by KerekesDavid** [comment]
The same is true for the new typed dictionaries, and not only for nested dicts, but also typed Arrays in Dictionaries are not allowed.

I have multiple use cases where allowing something like this would be very useful:
```gdscript
var pos_dict : Dictionary[Object, Array[Vector2i]]
```

**Comment by dalexeev** [comment]
* See also #12224.
